node/src/components/consensus/era_supervisor.rs

```diff
@@ -696,6 +697,28 @@ where
                 .collect(),
         }
     }
+
+    /// Emits a fatal error if the consensus state is still empty.
+    pub(super) fn shutdown_if_necessary(&self) -> Effects<Event<I>> {
+        let should_emit_error = self
+            .era_supervisor
+            .active_eras
+            .get(&self.era_supervisor.current_era)
+            .and_then(|era| {
+                let any_ref = era.consensus.as_any();
+                any_ref.downcast_ref::<HighwayProtocol<I, ClContext>>()
```

Comment:
Nooo! :weary: 
Please don't add another direct dependency on `HighwayProtocol` to `EraSupervisor`. I'm just trying to get rid of them in https://github.com/CasperLabs/casper-node/pull/475 so that I can write proper tests for it.

I'd very much prefer to add another method to the `ConsensusProtocol` trait.